lib/utils/circular_buffer_test.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2018-2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewCircularBuffer(t *testing.T) {
+	buff, err := NewCircularBuffer(-1)
+	require.Error(t, err)
+	require.Nil(t, buff)
+
+	buff, err = NewCircularBuffer(5)
+	require.NoError(t, err)
+	require.NotNil(t, buff)
+	require.Len(t, buff.buf, 5)
+}
+
+func TestCircularBuffer_Data(t *testing.T) {
```

Comment:
Consider making this test more comprehensive in order to catch the incorrect slice indexes. For example, see this updated [circular_buffer_test.go](https://github.com/gravitational/teleport/files/7238790/circular_buffer_test.go.txt) which is failing without the `t.end+1` fix suggested above.